### PR TITLE
Add k8s deployment files and github automatic build workflows

### PR DESCRIPTION
**Pull request by ramessesii2**
## Added k8s deployment files and github automatic build workflows
*  Docker file  for mks-server image
*  Add k8s config files for db and mks-server
*  Github workflows  for automatic build
* Minor fixes

## What's going to make deployment happen
We start with creating our mks resources in the cluster, then we apply the k8s db config files to get the redis-db server up and running, and finally we apply the mks-server config files which in return creates a service account having cluster role and  permissions to allow it to talk to k8s API server.
Finally, we have github workflows to add the ability to auto build and push the latest image to github registry.

## How to test
0. Get the cluster configured, and create a namespace "*mks*"
```bash
oc new-project mks
```
1. Install tekton pipelines operator in the cluster.
```bash
oc apply --filename https://storage.googleapis.com/tekton-releases/pipeline/latest/release.yaml
```
2. Apply the files to create the mks resources inside the cluster
```bash
oc apply -f config/mksCRDs
```
3 Get db server up and running.
```bash
oc apply -f k8s/db-config
```
4. Get mks-server up and running.
```bash
oc apply -f k8s/mks-config
```
5. See the logs to verify
```bash
oc logs -f -n mks mks-server-<get the deployment name>
```

### CODE REVIEW

File: .github/wokflows/release-package.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: mks-server image
```
Comment (avinal) [inline .github/wokflows/release-package.yaml:18]: Please add licence and copyright header to this file too

File: .github/wokflows/release-package.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ github.token }}
+
+      - name: Build mks-server image
+        run: docker  build . --tag $IMAGE_NAME
```
Comment (avinal) [inline .github/wokflows/release-package.yaml]:
```suggestion
        run: docker build . --tag $IMAGE_NAME
```

File: .github/wokflows/release-package.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+      - name: Push mks-server image to ghcr.io
+        run: docker  push $IMAGE_NAME
```
Comment (avinal) [inline .github/wokflows/release-package.yaml]:
```suggestion
        run: docker push $IMAGE_NAME
        
```

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ADD . /build/
+
+RUN mkdir /tmp/cache
+RUN CGO_ENABLED=0 GOCACHE=/tmp/cache go build -mod=vendor -v -o /tmp/mks-server .
+
+FROM scratch
+
+LABEL org.opencontainers.image.authors="Avinal Kumar <avinkuma@redhat.com>, Utkarsh Chaurasia <uchauras@redhat.com>, Satyam Bhardwaj <sabhardw@redhat.com>"
+LABEL org.opencontainers.image.source="https://github.com/MiniTeks/mks-server"
+LABEL org.opencontainers.image.version="1.0.1-beta"
```
Comment (avinal) [inline Dockerfile]:
```suggestion
LABEL org.opencontainers.image.version="latest"
```
Let's put the same version everywhere.

File: k8s/config/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      app: mks-server
+  template:
+    metadata:
+      labels:
+        app: mks-server
+    spec:
+      serviceAccountName: mks-server-sa
+      containers:
+      - name: mks-server
+        image: ghcr.io/ramessesii2/mks-server:latest
```
Comment (avinal) [inline k8s/config/deployment.yaml]:
```suggestion
        image: ghcr.io/MiniTeks/mks-server:latest
```

File: k8s/config/service.yaml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: mks-server
+  namespace: mks
+spec:
+  selector:
+    app: mks-server
+  externalIPs:
+    - "1.2.4.120"
```
Comment (avinal) [inline k8s/config/service.yaml]: Will this remain same for all deployments?

File: k8s/config/deployment.yaml (hunk 5 above)
Comment (ramessesii2) [reply k8s/config/deployment.yaml]: until we merge this branch with main, we won't find any builds inside MiniTeks/mks-server. So, for testing use ramessesii2/mks-server:latest. Although I'm gonna change it now

File: Dockerfile (hunk 4 above)
Comment (ramessesii2) [reply Dockerfile]: tq, done

File: k8s/config/service.yaml (hunk 6 above)
Comment (ramessesii2) [reply k8s/config/service.yaml]: yeah

File: .github/wokflows/release-package.yaml (hunk 1 above)
Comment (ramessesii2) [reply .github/wokflows/release-package.yaml:18]: ok, done

File: pkg/controllers/mkstask/controller.go [hunk 7]
```diff
@@ -97,7 +97,7 @@ func (c *controller) handleAdd(obj interface{}) {
 	if er != nil {
 		log.Fatalf("Cannot get tekton client: %s", er.Error())
 	}
-	tsk, err := Create(cs, obj.(*v1alpha1.MksTask), metav1.CreateOptions{}, "default")
+	tsk, err := Create(cs, obj.(*v1alpha1.MksTask), metav1.CreateOptions{}, obj.(*v1alpha1.MksTask).GetObjectMeta().GetNamespace())
```
Comment (avinal) [inline pkg/controllers/mkstask/controller.go]: `var addObj = obj.(*v1alpha1.MksTask)`

Use this for replacement.

**Review by UtkarshChaurasia — APPROVED** [review]
lgtm

File: pkg/controllers/mkstask/controller.go (hunk 7 above)
Comment (ramessesii2) [reply pkg/controllers/mkstask/controller.go]: tq, done

**Review by avinal — APPROVED** [review]
Great Work

**Pull request merged**